public/js/discover/discover.js

```diff
@@ -34,6 +34,12 @@ function updateSearchResults() {
   const filter = document.getElementById('search-dropdown').value;
   fetch('/discover/' + filter)
     .then((data) => data.json())
+    .then(async (organizations) => {
+      for (const organization of organizations) {
+        await setLocationInfo(organization);
+      }
```

Comment:
I think this might wait on each one individually when you could resolve the promises in parallel. 

Might be able to use ```Promise.all``` here. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all